sdk/webpubsub/azure-webpubsub-client/azure/webpubsub/client/_client.py

```diff
@@ -0,0 +1,864 @@
+        if self._protocol.is_reliable_sub_protocol and (
+            (self._thread_seq_ack and not self._thread_seq_ack.is_alive()) or (self._thread_seq_ack is None)
+        ):
+
+            def sequence_id_ack_periodically():
+                while self._is_connected():
+                    try:
+                        is_updated, seq_id = self._sequence_id.try_get_sequence_id()
+                        if is_updated and seq_id is not None:
+                            self._send_message(SequenceAckMessage(sequence_id=seq_id))
+                    finally:
+                        delay(1.0)
+
+            self._thread_seq_ack = threading.Thread(target=sequence_id_ack_periodically, daemon=True)
+            self._thread_seq_ack.start()
+
+    def _start_core(self):
+        self._state = WebPubSubClientState.CONNECTING
+        _LOGGER.info("Staring a new connection")
+
+        # Reset before a pure new connection
+        self._sequence_id.reset()
+        self._is_initial_connected = False
+        self._last_close_event = None
+        self._last_disconnected_message = None
+        self._connection_id = None
+        self._reconnection_token = None
+        self._url = None
+
+        self._url = self._credential.get_client_access_url()
+        self._connect(self._url)
+
+    @distributed_trace
+    def _start(self) -> None:
+        """start the client and connect to service"""
+
+        if self._is_stopping:
+            raise Exception("Can't start a client during stopping")
+        if self._state != WebPubSubClientState.STOPPED:
+            raise Exception("Client can be only started when it's Stopped")
+
+        try:
+            self._start_core()
+        except Exception as e:
+            self._state = WebPubSubClientState.STOPPED
+            self._is_stopping = False
+            raise e
+
+    @distributed_trace
+    def _stop(self) -> None:
+        """stop the client"""
+
+        if self._state == WebPubSubClientState.STOPPED or self._is_stopping:
+            return
+        self._is_stopping = True
+
+        old_thread = self._thread
+        old_thread_seq_ack = self._thread_seq_ack
+
+        # we can't use self._ws.close otherwise on_close may not be triggered
+        # (realted issue: https://github.com/websocket-client/websocket-client/issues/899)
+        if self._ws and self._ws.sock:
+            self._ws.sock.close()
+
+        # users may call start the client again after stop so we need to wait for old thread join
+        if old_thread_seq_ack and old_thread_seq_ack.is_alive():
+            _LOGGER.debug("wait for seq thread stop")
+            old_thread_seq_ack.join()
+        if old_thread and old_thread.is_alive():
+            _LOGGER.debug("wait for listener thread stop")
+            old_thread.join()
+
+        _LOGGER.info("stop client successfully")
+
+    @overload
+    def on(
+        self, event: Union[Literal[CallBackType.CONNECTED], str], listener: Callable[[OnConnectedArgs], None]
```

Comment:
I don't think you need to have a `Union` type here - the `Literal` is already `str`, so it's implied - and I think it will confuse the intellisense experience.

Reply:
updated